core/trino-main/src/main/java/io/trino/FeaturesConfig.java

```diff
@@ -479,6 +482,20 @@ public FeaturesConfig setAllowSetViewAuthorization(boolean allowSetViewAuthoriza
         return this;
     }
 
+    @NotNull
+    public Duration getMaterializedViewRequiredFreshness()
```

Comment:
I think this is very much property of data, not a system wide things. I also think refresh scheduler should be outside of core engine and engine should only be responsible for executing refresh or select queries.

I don't think that engine needs to know freshness duration if it cannot do much about if without a scheduler.

High level question is:
* Do we need resolution as MV as view at all? 
  * If no, then the whole concept of freshness doesn't make sense.
  * If yes, then IMO it's the connector that should be responsible for business (time-based) logic whether MV is fresh or not. For example freshness might be tightly coupled with refresh interval and MV can be refreshed based on regex (e.g. every day on midnight).